src/IsomorphicRequest.ts

```diff
@@ -0,0 +1,57 @@
+import { Headers } from 'headers-polyfill/lib'
+import { decodeBuffer } from './utils/bufferUtils'
+import { uuidv4 } from './utils/uuid'
+
+export interface RequestInit {
+  method?: string
+  headers?: Record<string, string | string[]> | Headers
+  credentials?: RequestCredentials
+  body?: ArrayBuffer
+}
+
+export class IsomorphicRequest {
+  public id: string
```

Comment:
Need to keep `id` as writable because our test helpers re-construct `IsomorphicRequest` instance using an intercepted request object in Playwright.